1-4 задачи решать с помощью функций, 5ю вручную

Выбрать тест и проверить, есть  ли различия между выборками:

Задача №1  
Даны две  независимые выборки. Не соблюдается условие нормальности

x1  380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

In [12]:
import scipy.stats as st
import numpy as np

x1 = np.array([380,420, 290])
x2 = np.array([140,360,200,900])
res1 = st.mannwhitneyu(x1, x2)
res1

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Вывод: Статистически значимых различий при уровне значимости 5% не обнаружено.

Задача №2  
Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

In [13]:
z1 = np.array([150, 160, 165, 145, 155])
z2 = np.array([140, 155, 150, 130, 135])
z3 = np.array([130, 130, 120, 130, 125])
res2 = st.friedmanchisquare(z1, z2, z3)
res2

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Вывод: статистически значимые различия есть, препарат влияет на уровень давления.

Задача №3  
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [14]:
res3 = st.wilcoxon(z1, z2)
res3

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Вывод: Статистически значимых различий не обнаружено, препарат не влияет на уровень давления.

Задача №4  
Даны 3 группы  учеников плавания.

В 1 группе время на дистанцию 50 м составляют: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [15]:
c1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
c2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
c3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
res4 = st.kruskal(c1, c2, c3)
res4

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

Вопроса в задаче нет! Вывод: Статистически значимых различий не обнаружено. 

Задача №5  
Заявляется, что партия изготавливается со средним арифметическим 2,5 см. 
Проверить данную гипотезу, если известно, что размеры изделий подчинены 
нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [47]:
m = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
sr = np.mean(m)
sr

2.5279999999999996

In [57]:
alf = (1 - 0.05)/2
alf

0.475

In [58]:
mu = np.std(m, ddof = 1)
mu

0.1572542173961923

In [59]:
i = len(m)
t = (sr - 2.5)/(mu/np.sqrt(i))
t

0.5630613661802959

In [60]:
k = st.t.ppf(alf, (i-1))
k

-0.06447679010158362

In [61]:
l = 1 - 2*(1 - st.t.cdf(t, i-1))
l

0.4128560006059372

In [62]:
pv = 1 + l
pv

1.4128560006059372

Вывод: Наблюдаемое значение 0.563 попадает в интервал двухстороннего теста от -0.644 до 0.644 следовательно это нулевая гипотеза (что партия изготавливается со средним арифметическим 2,5 см)